includes/class-recaptcha.php

```diff
@@ -286,6 +298,58 @@ public static function verify_captcha( $captcha_token ) {
 
 		return true;
 	}
+
+	/**
+	 * Hook reCAPTCHA v3 to WooCommerce.
+	 */
+	public static function woocommerce_hooks() {
+		// Add reCAPTCHA v3 to the checkout form.
+		\add_action( 'woocommerce_after_checkout_form', [ __CLASS__, 'add_recaptcha_to_checkout' ] );
+
+		// Verify reCAPTCHA v3 on checkout submission.
+		\add_action( 'woocommerce_checkout_process', [ __CLASS__, 'verify_recaptcha_on_checkout' ] );
+	}
+
+	/**
+	 * Add reCAPTCHA v3 to checkout.
+	 */
+	public static function add_recaptcha_to_checkout() {
+		if ( ! self::can_use_captcha() ) {
+			return;
+		}
+		$settings = self::get_settings();
```

Comment:
Nit: we can use `get_setting( 'site_key' )` here as a shortcut (since the options exist as individual rows and there's no need to fetch the other settings).

Reply:
Good call!